### ISSUE TITLE
apns-expiration

### ISSUE THREAD

**Issue by sisoje** [AndrewBarba/apns2#18]
Hello,
I dont understand the following code? Why dividing with 1000? And why parseInt AFTER dividing already? 

```
if (notification.expiration) {
      options.headers['apns-expiration'] = parseInt(notification.expiration / 1000)
}
```

**Comment by AndrewBarba** [comment]
Timestamps in Javascript are reported in milliseconds, Apple wants the expiration reported in seconds as an integer. So we dived the javascript timestamp by 1000 and round to an int. `parseInt` can take in any number of string. When the input is a Number it will always return an integer but with no decimal precision

**Comment by sisoje** [comment]
you should not divide by 1000 deep inside the code, because its hidden from the user. its a bad practice, and its confusing. I understand dividing by 1000 is a bit of boilerplate code for javascript devs, but boilerplate code should be kept outside of the core functionality.

if i have many notifications with same expiration then setting each of them will cause unnecessary division inside.

also, not all timestamps will come from javascript. c time() gives seconds already (same in PHP). 

its a value clearly described by apple in seconds and should be given as such regardless of the language used.

FYI I am not javascript developer, I just make payload and parameters to be consumed by this library.

**Comment by sisoje** [comment]
huh now i see...
its best to change APNS constructor to receive default headers, instead like now its receiving `defaultTopic`, and at the end we have half of the properties in notification and half in the apns... a bit messy

**Comment by sisoje** [comment]
ok forked and done :)

**Comment by sisoje** [comment]
i appreciate your work.

the problem is that i use azure functions and i send messages in ~500 tokens at a time and all have the same payload and same headers and for each batch i create a new client because I don't use "long lived" instances, so, sadly, i could not really completely fit into your library... also I dont trust bluebirds `Promise.map` some people complain it stuck on big batches....

thats why i made a simple fork to do what i need. so i set everything "RAW" like `apns-expiration` and `apns-collapse-id` and do that only once per batch in the common header. also payload is RAW once per batch.

TIP1 it seems that your framework does not provide `apns-id` from the result (and possibly not even for the input), maybe someone will need it?

TIP2 for me seems more natural to provide a `production` flag instead of entire host to the constructor

**Comment by AndrewBarba** [comment]
Cool makes sense, thanks for the tips. Beauty of open source is you can fork and adjust for your needs, glad you were able to get it working.

**Comment by AndrewBarba** [comment]
I just released 6.0.0: https://github.com/AndrewBarba/apns2/releases
